x/accounts/module.go

```diff
@@ -42,21 +43,29 @@ type AppModule struct {
 	k   Keeper
 }
 
-func (m AppModule) IsAppModule() {}
+func (AppModule) IsAppModule() {}
 
 // Name returns the module's name.
 // Deprecated: kept for legacy reasons.
-func (AppModule) Name() string { return ModuleName }
+func (am AppModule) Name() string { return ModuleName }
+
+func (AppModule) RegisterInterfaces(registrar registry.InterfaceRegistrar) {
+	registrar.RegisterImplementations((*coretransaction.Msg)(nil),
+		&basev1.MsgInit{},
```

Comment:
I've registered the types manually because this method https://github.com/cosmos/cosmos-sdk/blob/fc91e76354512e8538f9da5bc9d3518f54921417/x/accounts/keeper.go#L442 doesn't work. 
Why do we do dynamic registration? If it makes sense we could try to fix it and if not add to the manual registration the missing ones as we have in the other modules.

Reply:
This is not correct, i think, we should not register this here because it assumes x/accounts knows the accounts being plugged, which is not true

Reply:
yeah this should be removed. What is the PR that going to fix this ? Maybe then we can merge that PR first then this after 🤔